addons/stock/tests/test_proc_rule.py

```diff
@@ -19,6 +19,66 @@ def setUp(self):
         })
         self.partner = self.env['res.partner'].create({'name': 'Partner'})
 
+    def test_orderpoint_replenishment_view(self):
+        """ Create two warehouses + a sale order
+        verify that the replenishment view is consistent"""
+        warehouse_1 = self.env['stock.warehouse'].search([('company_id', '=', self.env.company.id)], limit=1)
+        warehouse_2 = self.env['stock.warehouse'].create({
+            'name': 'Warehouse Two',
+            'code': 'WH2',
+            'resupply_wh_ids': [warehouse_1.id],
+        })
+        warehouse_3 = self.env['stock.warehouse'].create({
+            'name': 'Warehouse Three',
+            'code': 'WH3',
+            'resupply_wh_ids': [warehouse_1.id],
+        })
+        route_2 = self.env['stock.location.route'].search([
+            ('supplied_wh_id', '=', warehouse_2.id),
+            ('supplier_wh_id', '=', warehouse_1.id),
+        ])
+        route_3 = self.env['stock.location.route'].search([
+            ('supplied_wh_id', '=', warehouse_3.id),
+            ('supplier_wh_id', '=', warehouse_1.id),
+        ])
+        product = self.env['product.product'].create({
+            'name': 'Super Product',
+            'type': 'product',
+            'route_ids': [route_2.id, route_3.id]
+        })
+        move1 = self.env['stock.move'].create({
+            'name': 'Move WH2',
+            'location_id': route_2.supplied_wh_id.lot_stock_id.location_id.id,
+            'location_dest_id': self.partner.property_stock_customer.id,
+            'product_id': product.id,
+            'product_uom': product.uom_id.id,
+            'product_uom_qty': 1,
+        })
+        move1._action_confirm()
+        move2 = self.env['stock.move'].create({
+            'name': 'Move WH2',
+            'location_id': route_3.supplied_wh_id.lot_stock_id.location_id.id,
+            'location_dest_id': self.partner.property_stock_customer.id,
+            'product_id': product.id,
+            'product_uom': product.uom_id.id,
+            'product_uom_qty': 1,
+        })
+        move2._action_confirm()
+        # activate action of opening the replenishment view
+        self.env['report.stock.quantity'].flush()
+        Orderpoint = self.env['stock.warehouse.orderpoint']
+        Orderpoint.action_open_orderpoints()
+        replenishment = self.env['stock.warehouse.orderpoint'].search([
+            ('product_id', '=', product.id),
+        ])
+        # check that the location's wharehouse is the warehouse supplied in "preferred route"
+        self.assertEqual(replenishment[0].route_id.supplied_wh_id.lot_stock_id,
```

Comment:
It's a bit weird not to check the `route_id` (which is the fix goal)
Something like this may be clearer and more relevant
```suggestion
        self.assertRecordValues(replenishments, [
            {'location_id': warehouse_2.lot_stock_id.id, 'route_id': route_2.id},
            {'location_id': warehouse_3.lot_stock_id.id, 'route_id': route_3.id},
        ])
```